File: WorkoutTracker.Domain/Exercises/ExerciseInWorkout.cs
using System;
using System.Collections.Generic;
using System.Text;
using WorkoutTracker.Domain.BaseClasses;

namespace WorkoutTracker.Domain.Exercises
{
    public class ExerciseInWorkout : Entity
    {
        public virtual Exercise Exercise { get; set; }
        public virtual int ExerciseId { get; set; }
        public byte NumberOfSets { get; set; }
        public SetType SetType { get; set; }
        public byte Sequence { get; set; }
    }
}

File: workout-tracker/src/app/workouts/_models/exercise-in-workout.ts
import { Entity } from 'app/shared/models/entity';
import { Exercise } from './exercise';

export class ExerciseInWorkout extends Entity {

  public exercise: Exercise | null;
  public exerciseId: number;
  public numberOfSets: number;
  public setType: number;
  public sequence: number;

  constructor(
    id: number,
    exerciseId: number,
    exerciseName: string,
    numberOfSets: number,
    setType: number,
    sequence: number) {
    super();
    this.id = id;
    this.exerciseId = exerciseId;
    this.exercise = new Exercise();
    this.exercise.name = exerciseName;
    this.numberOfSets = numberOfSets;
    this.setType = setType;
    this.sequence = sequence;
    this.exercise = null;
  }
}
